### PR TITLE
Ensure tasks are resolved when run from extension host

### LINKED ISSUE #101884: vscode.tasks.executeTask ignores dependsOn
Issue Type: <b>Bug</b>

This seems to be a regression on VS Code Insiders. It does _not_ repro on stable. If I call `vscode.tasks.executeTask`, it will only execute that specific task, not the task it dependsOn as well. Here's an example tasks.json:
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "echo",
            "type": "shell",
            "dependsOn": "echo2",
            "command": "echo Hello",
            "problemMatcher": []
        },
        {
            "label": "echo2",
            "type": "shell",
            "command": "echo Hello2",
            "problemMatcher": []
        }
    ]
}
```

Example code to execute the task:
```
const tasks = await vscode.tasks.fetchTasks();
await vscode.tasks.executeTask(tasks[0]);
```

### Expected
```
> Executing task: echo Hello2 <
Hello2
Terminal will be reused by tasks, press any key to close it.
> Executing task: echo Hello <
Hello
Terminal will be reused by tasks, press any key to close it.
```

### Actual:
```
> Executing task: echo Hello <
Hello
Terminal will be reused by tasks, press any key to close it.
```

VS Code version: Code - Insiders 1.47.0-insider (e95f9e63380a139cb1cbe54bacfc6703b96e30b9, 2020-07-07T07:46:45.605Z)
OS version: Darwin x64 19.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.05GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-pull-request-github|Git|0.17.0
vscode-azureappservice|ms-|0.17.1-alpha
vscode-azureresourcegroups|ms-|0.2.0
vscode-azurestorage|ms-|0.9.0
vscode-cosmosdb|ms-|0.13.0
csharp|ms-|1.22.1
python|ms-|2020.6.91350
remote-containers|ms-|0.127.0
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
azure-account|ms-|0.8.11
powershell|ms-|2020.6.0
vscode-typescript-tslint-plugin|ms-|1.2.3
java|red|0.64.0
vscode-didact|red|0.1.15
sort-lines|Tyr|1.9.0
vscode-java-debug|vsc|0.26.0

</details>

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #101884

### CODE REVIEW

**Comment by alexr00** [comment]
This issue was caused by https://github.com/Microsoft/vscode/commit/1abf21d3d80d604bd2d4f5b402bdcbb04e88780a.
https://github.com/Microsoft/vscode/commit/1abf21d3d80d604bd2d4f5b402bdcbb04e88780a fixed some timing issues with running tasks from the extension host by making an extra trip from the extension host to main to get the task execution, essentially splitting mainThreadTasks `$executeTask` into `$getTaskExecution` (called first) and `$executeTask` (called second).

This split is what caused #101884. When a task execution is made, the task is effectively resolved. So, in the old way with just `$executeTask`, the task was resolved and then immediately run. Now, the task is resolved, passed to the extension host where it loses some important stuff (like the depends on info), passed back to the main thread, then executed. There needs to be another resolve in there before it is run, so that it can pick up the missing depends on info again.

In short, I have changed the one place where a `TaskHandleDTO` is passed into `$executeTask` to pass that again, and I have changed `$executeTask` back to it's pre-timing fix behavior. This still preserves the timing fixes, since `$getTaskExecution` is still being called first. The changes to `$executeTask` that caused this issue were not to fix the timing problems, but instead to increase performance by not resolving the task again. `$executeTask` and `$getTaskExecution` should look very similar, with `$executeTask` performing the additional `run`.

**Review by dbaeumer — APPROVED** [review]
